libp2p/peer/id.py

```diff
@@ -13,64 +20,68 @@
 
 class ID:
 
-    def __init__(self, id_str):
+    _id_str: str
+
+    def __init__(self, id_str: str) -> None:
         self._id_str = id_str
 
-    def get_raw_id(self):
+    def get_raw_id(self) -> str:
         return self._id_str
 
-    def pretty(self):
+    def pretty(self) -> str:
         return base58.b58encode(self._id_str).decode()
 
-    def get_xor_id(self):
+    def get_xor_id(self) -> int:
         return int(digest(self.get_raw_id()).hex(), 16)
 
-    def __str__(self):
+    def __str__(self) -> str:
         pid = self.pretty()
         return pid
```

Comment:
Just a random thought: Since we have `__str__`, what do you think if we also add a `__bytes__`, which makes us able to do  `bytes(peer_id)`?

Reply:
IMO sticking to one way of getting ID in bytes would be less confusing, either `ID.get_raw_id()` or `bytes(peer_id)` works for me.